test/rest-router.test.js

```diff
@@ -0,0 +1,60 @@
+var assert = require('assert');
+var expect = require('chai').expect;
+var RestRouter = require('../lib/rest-router');
+
+describe('Custom Router', function() {
+  var restRouter;
+  var route = { verb: 'get', fullPath: '/Planets' };
+  var handler = function(req, res, next) { res.handled = true; };
+  var req = {
+    method: 'get',
+    url: '/Planets',
+  };
+  var res = {};
+  var next = function() { res.calledWhenPathNotFound = true; };
+
+  beforeEach(function() {
+    restRouter = new RestRouter();
+  });
+
+  describe('RestRouter()', function() {
+
+    it('initiates a restRouter obj with Trie and options', function() {
+      expect(restRouter).to.have.keys(['trie', 'options']);
+    });
+  });
+
+  describe('restRouter.registerPathAndHandlers()', function() {
+    it('register handler for GET: /Planets', function() {
+      restRouter.registerPathAndHandlers(route, handler);
+      expect(restRouter.trie).to.have.deep.property('planets.methods.get');
+    });
+  });
+
+  describe('restRouter.handle()', function() {
+    it('returns a function --which invokes matching handler-- with three arguments',
+      function() {
+        restRouter.registerPathAndHandlers(route, handler);
+        var returned = restRouter.handle();
+        expect(returned).to.have.length(3);
+      });
```

Comment:
Instead of asserting on function API, it's better to invoke the handler via express as I described above. That way we verify that the router actually works with express, instead of verifying that it matches an API that we think will work.